### PR TITLE
Feature/KYV-377 recreate subscription

### PR DESCRIPTION
**Pull request by wpamb**

### CODE REVIEW

File: packages/order-backend/src/subscription/types.ts [hunk 1]
```diff
@@ -31,4 +32,20 @@ export interface Subscription {
   priceGross: string
   vatPercentage: string
   orderId: string
+  originalPriceNet: string
+  originalPriceVat: string
+  originalPriceGross: string
+  meta: SubscriptionItemMeta[]
```
Comment (severikupari1) [inline packages/order-backend/src/subscription/types.ts:38]: Is meta optional?

**Review by severikupari1 — CHANGES_REQUESTED** [review]

**Comment by severikupari1** [comment]
Can you create automation tests for this case functional/unit?

File: packages/order-backend/src/subscription/types.ts (hunk 1 above)
Comment (wpamb) [reply packages/order-backend/src/subscription/types.ts:38]: Shouldn't be

File: packages/order-backend/src/subscription/types.ts [hunk 2]
```diff
@@ -31,4 +32,20 @@ export interface Subscription {
...
+}
+
+export interface SubscriptionItemMeta {
+  orderItemMetaId: string
+  orderItemId: string
+  orderId: string
+  subscriptionId: string
+  key: string
+  value: string
+  label: string
```
[the remark this answers is not in the document]
Comment (wpamb) [reply packages/order-backend/src/subscription/types.ts:48]: Might be. The typings in general are quite lacking with a lot of values possibly missing etc etc

File: packages/order-experience-api/src/api/recreateSubscription.ts [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+    if (status !== 'active') {
+      throw new ExperienceError({
+        code: 'subscription-validation-failed',
+        message: `subscription ${subscriptionId} must be active to recreate`,
+        logLevel: 'debug',
+        responseStatus: StatusCode.BadRequest,
+      })
+    }
+
+    const newSubscription = await createSubscription({
```
[the remark this answers is not in the document]
Comment (wpamb) [reply packages/order-experience-api/src/api/recreateSubscription.ts:90]: That's a good idea. I'll fix this

File: packages/order-experience-api/src/api/recreateSubscription.ts [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+      ...oldSubscription,
+      ...updatedFields,
+      startDate: new Date().toISOString(),
+      endDate: oldSubscription.endDate,
+      billingStartDate: oldSubscription.endDate,
```
[the remark this answers is not in the document]
Comment (wpamb) [reply packages/order-experience-api/src/api/recreateSubscription.ts:95]: The ticket at least has specified it as old subscription's endDate

**Review by severikupari1 — APPROVED** [review]

**Pull request merged**